Lib/ntpath.py

```diff
@@ -693,30 +688,31 @@ def _getfinalpathname_nonstrict(path):
                 path = _getfinalpathname(path)
                 return join(path, tail) if tail else path
             except OSError as ex:
-                if ex.winerror not in allowed_winerror:
+                winerror = ex.winerror
+                if winerror not in allowed_winerror:
                     raise
+            try:
```

Comment:
I understand that the old code didn't _need_ to be nested under the outer `except:`, given that the outer `try:` block above always returns.

But by "unintenting" it here, are you actually making anything faster? If not, please revert this. I think Alex has already listed some reasons why we don't accept these sorts of changes, but to name a couple of my own:

- CPython is a mature software project, and so reviewers are most interested in the _risk_ that PRs introduce vs the potential improvements. Every line of code changed carries some risk unless reviewed in great detail. Any sort of rewrite is viewed with great suspicion. We'd rather keep the bugs and code smells we have than introduce unknown defects for users by mistake.
- Stylistic changes make the `git blame` of the file less useful
- Stylistic changes make backports more likely to hit conflicts that require manual resolution

Reply:
I believe it should make the traceback shorter. As no exception will occur during the handling of the error.

Reply:
> I believe it should make the traceback shorter. As no exception will occur during the handling of the error.

In that case, it sounds like a separate feature that needs tests, and probably deserves its own PR, rather than being bundled together with the performance-focussed changes in this PR

Reply:
Ah OK. But that sounds like it's outside the scope of this PR and this issue, right?

Reply:
And if I use an if statement to indent it?

Reply:
> And if I use an if statement to indent it?

This is borderline "malicious compliance" and I am quite close to unsubscribing from this PR. Why are you taking this approach, e.g. trying to find ways around what's being asked, or not removing unrelated changed unless each one is individually called out?

Reply:
> And if I (ab)use an if statement to indent it?

I think that's just going to be really perplexing for people reading the code in the future, even with the comment you added :-)

If it's worth making the change because it leads to a cleaner traceback for users, then we should make the change, but in a separate PR rather than this one, since the change has nothing to do with performance improvements

Reply:
I'm extremely sorry, I should have simply moved it once again to a different pull request.
I'm NOT trying to find ways around what's being asked:

- I sometimes simply ask for clarification: https://github.com/python/cpython/pull/117350#discussion_r1544699816
- Answer questions asked: https://github.com/python/cpython/pull/117350#discussion_r1544745554, but I didn't bother to explicitely mention that I would revert it, which I did.

I'm trying to remove as many unrelated changed even if they aren't individually called out, because they'll probably be called out if I don't, but I look over some of them.
I usually have whitespace differences disabled in the diff viewer, precisely because it's difficult to see what I changed...

Reply:
Hmm, I upon closer inspection, it probably can't raise an exception.